src/xviewer-image.c

```diff
@@ -720,6 +716,7 @@ xviewer_image_get_file_info (XviewerImage *img,
             GFileInputStream *input_stream;
             guchar ip_buffer [BUFFER_SIZE];
             gint bytes_read;
+            char *mime_type_by_extn;
 
```

Comment:
It looks like `mime_type` is never initialized inside of this scope, and isn't guaranteed to be set, maybe `*mime_type = NULL;` here?

Reply:
I think that this is OK as it stands.

mime_type is a calling parameter that is set to something other than NULL if the function is to return the MIME type of the file. (The parameter bytes is similarly non-NULL if the size of the file is required.

mime_type_by_extn is always set before it is used - it could be initialised to NULL but there is little point.